apps/frontend/src/pages/ViewListing.tsx

```diff
@@ -1,5 +1,128 @@
+import {
+  Button,
+  Card,
+  CardContent,
+  Container,
+  Grid,
+  Link,
+  Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import _axios_instance from "../_axios_instance.tsx";
+import Carousel from "../extra_components/Carousel.tsx";
+import { useStyles } from "../styles/pageStyles.tsx";
+import { colors } from "../styles/colors.tsx";
+
 const ViewListing = () => {
-  return <h1>ViewListing</h1>;
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const [listingObject, setListingObject] = useState({
+    title: "Sample Title",
+    description: "Sample Description",
+    price: 0,
+    seller_profile: { name: "John Smith" },
+    dateCreated: "2024-05-23T15:30:00Z",
+    distance: 4.2,
+    images: [
+      { url: "https://picsum.photos/1200/400" },
+      { url: "https://picsum.photos/1200/600" },
+      { url: "https://picsum.photos/1200/1200" },
+      { url: "https://picsum.photos/600/1200" },
+      { url: "https://picsum.photos/400/1200" },
+    ],
+  });
+
+  // Load the listing from the api given an ID
+  useEffect(() => {
+    _axios_instance
+      .get("/listing/" + id)
+      .then((response) => {
+        setListingObject(response.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, [id]);
+
+  // Convert price to string
+  const priceFormatter = new Intl.NumberFormat("en-CA", {
+    style: "currency",
+    currency: "CAD",
+  });
+
+  // Convert date to string
+  const convertDate = (dateString: string) => {
+    const dateTimeObject = new Date(dateString);
+    return dateTimeObject.toDateString();
+  };
+
+  // TODO Make the routing with auth work properly
+  const handleNavToMessages = () => {
+    navigate("/messages");
+  };
+
+  return (
+    <Container>
+      <Card sx={{ height: "100%", width: "100%" }}>
+        <CardContent>
+          <Grid container spacing={2}>
+            <Grid item sm={12} md={12} lg={6}>
+              <Typography variant={"h2"}>
+                {listingObject !== null
```

Comment:
`listingObject?.title ?? "Title not received"`

the ?? operator returns the left hand side unless it is null or undefined, in which case it returns the right hand side. This also misses the case where listingObject is defined but listingObject.title is null which could lead to an unhandled runtime error